In [19]:
# Import Libraries

In [21]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [22]:
# Load the datasets

In [23]:
matches = pd.read_csv("matches.csv")
deliveries = pd.read_csv("deliveries.csv")

print("Shape of the matches dataset: ",matches.shape)
print("Shape of the deliveries dataset: ",deliveries.shape)



Shape of the matches dataset:  (1095, 20)
Shape of the deliveries dataset:  (260920, 17)


In [25]:
# Quick sanity check

In [67]:
print("Shape of the matches dataset: ",matches.shape)

print("Shape of the deliveries dataset: ",deliveries.shape)

print("First 5 rows of the matches dataset: ",matches.head())

print("\nAll columns in matches dataset:", matches.columns.tolist())

Shape of the matches dataset:  (1095, 20)
Shape of the deliveries dataset:  (260920, 17)
First 5 rows of the matches dataset:         id   season        city        date match_type player_of_match  \
0  335982  2007/08   Bangalore  2008-04-18     League     BB McCullum   
1  335983  2007/08  Chandigarh  2008-04-19     League      MEK Hussey   
2  335984  2007/08       Delhi  2008-04-19     League     MF Maharoof   
3  335985  2007/08      Mumbai  2008-04-20     League      MV Boucher   
4  335986  2007/08     Kolkata  2008-04-20     League       DJ Hussey   

                                        venue                        team1  \
0                       M Chinnaswamy Stadium  Royal Challengers Bengaluru   
1  Punjab Cricket Association Stadium, Mohali              Kings XI Punjab   
2                            Feroz Shah Kotla             Delhi Daredevils   
3                            Wankhede Stadium               Mumbai Indians   
4                                Eden Garden

In [ ]:
# Check whether any single franchise with different names

In [33]:
pd.concat([matches['team1'], matches['team2']]).unique() 

<StringArray>
['Royal Challengers Bangalore',             'Kings XI Punjab',
            'Delhi Daredevils',              'Mumbai Indians',
       'Kolkata Knight Riders',            'Rajasthan Royals',
             'Deccan Chargers',         'Chennai Super Kings',
        'Kochi Tuskers Kerala',               'Pune Warriors',
         'Sunrisers Hyderabad',               'Gujarat Lions',
     'Rising Pune Supergiants',      'Rising Pune Supergiant',
              'Delhi Capitals',                'Punjab Kings',
        'Lucknow Super Giants',              'Gujarat Titans',
 'Royal Challengers Bengaluru']
Length: 19, dtype: str

In [36]:
# As we can see a franchise with two names: Royal challengers Bangalore and Royal challenegers bengaluru, we replace it with single name

In [50]:
matches.replace(
    {
        'Royal Challengers Bangalore': 'Royal Challengers Bengaluru'
    },
    inplace=True
)

,id,season,city,date,match_type,player_of_match,venue,team1,team2,toss_winner,toss_decision,winner,result,result_margin,target_runs,target_overs,super_over,method,umpire1,umpire2
0,335982,2007/08,Bangalore,2008-04-18,League,BB McCullum,M Chinnaswamy Stadium,Royal Challengers Bengaluru,Kolkata Knight Riders,Royal Challengers Bengaluru,field,Kolkata Knight Riders,runs,140.0,223.0,20.0,N,NaN,Asad Rauf,RE Koertzen
1,335983,2007/08,Chandigarh,2008-04-19,League,MEK Hussey,"Punjab Cricket Association Stadium, Mohali",Kings XI Punjab,Chennai Super Kings,Chennai Super Kings,bat,Chennai Super Kings,runs,33.0,241.0,20.0,N,NaN,MR Benson,SL Shastri
2,335984,2007/08,Delhi,2008-04-19,League,MF Maharoof,Feroz Shah Kotla,Delhi Daredevils,Rajasthan Royals,Rajasthan Royals,bat,Delhi Daredevils,wickets,9.0,130.0,20.0,N,NaN,Aleem Dar,GA Pratapkumar
3,335985,2007/08,Mumbai,2008-04-20,League,MV Boucher,Wankhede Stadium,Mumbai Indians,Royal Challengers Bengaluru,Mumbai Indians,bat,Royal Challengers Bengaluru,wickets,5.0,166.0,20.0,N,NaN,SJ Davis,DJ Harper
4,335986,2007/08,Kolkata,2008-04-20,League,DJ Hussey,Eden Gardens,Kolkata Knight Riders,Deccan Chargers,Deccan Chargers,bat,Kolkata Knight Riders,wickets,5.0,111.0,20.0,N,NaN,BF Bowden,K Hariharan
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1090,1426307,2024,Hyderabad,2024-05-19,League,Abhishek Sharma,"Rajiv Gandhi International Stadium, Uppal, Hyd...",Punjab Kings,Sunrisers Hyderabad,Punjab Kings,bat,Sunrisers Hyderabad,wickets,4.0,215.0,20.0,N,NaN,Nitin Menon,VK Sharma
1091,1426309,2024,Ahmedabad,2024-05-21,Qualifier 1,MA Starc,"Narendra Modi Stadium, Ahmedabad",Sunrisers Hyderabad,Kolkata Knight Riders,Sunrisers Hyderabad,bat,Kolkata Knight Riders,wickets,8.0,160.0,20.0,N,NaN,AK Chaudhary,R Pandit
1092,1426310,2024,Ahmedabad,2024-05-22,Eliminator,R Ashwin,"Narendra Modi Stadium, Ahmedabad",Royal Challengers Bengaluru,Rajasthan Royals,Rajasthan Royals,field,Rajasthan Royals,wickets,4.0,173.0,20.0,N,NaN,KN Ananthapadmanabhan,MV Saidharshan Kumar
1093,1426311,2024,Chennai,2024-05-24,Qualifier 2,Shahbaz Ahmed,"MA Chidambaram Stadium, Chepauk, Chennai",Sunrisers Hyderabad,Rajasthan Royals,Rajasthan Royals,field,Sunrisers Hyderabad,runs,36.0,176.0,20.0,N,NaN,Nitin Menon,VK Sharma


In [51]:
# Select the team and Filter the team matches

In [63]:
My_Team = 'Royal Challengers Bengaluru'

rcb_matches = matches[ (matches['team1'] == My_Team) | (matches['team2'] == My_Team )]

print(f"Total IPL matches in dataset: {len(matches)}")
print(f"Total no of matches played by RCB: {len(rcb_matches)}")
print(f"RCB has played in {(len(rcb_matches)/len(matches))*100 :.1f}% of all IPL matches")

Total IPL matches in dataset: 1095
Total no of matches played by RCB: 255
RCB has played in 23.3% of all IPL matches


In [64]:
#Quick look at their most recent matches

In [66]:
print(rcb_matches[['season', 'team1', 'team2', 'winner', 'venue']].tail(5))

     season                        team1                        team2  \
1075   2024               Gujarat Titans  Royal Challengers Bengaluru   
1081   2024  Royal Challengers Bengaluru                 Punjab Kings   
1085   2024  Royal Challengers Bengaluru               Delhi Capitals   
1089   2024  Royal Challengers Bengaluru          Chennai Super Kings   
1092   2024  Royal Challengers Bengaluru             Rajasthan Royals   

                           winner  \
1075  Royal Challengers Bengaluru   
1081  Royal Challengers Bengaluru   
1085  Royal Challengers Bengaluru   
1089  Royal Challengers Bengaluru   
1092             Rajasthan Royals   

                                                  venue  
1075                   M Chinnaswamy Stadium, Bengaluru  
1081  Himachal Pradesh Cricket Association Stadium, ...  
1085                   M Chinnaswamy Stadium, Bengaluru  
1089                   M Chinnaswamy Stadium, Bengaluru  
1092                   Narendra Modi Stadium, Ah

In [81]:
# Data Quality Check

In [95]:
# Check missing values in RCB's match data
missing_values = rcb_matches.isnull().sum()
missing_percentage = missing_values / len(rcb_matches)

# Create a summary dataframe
missing_df = pd.DataFrame({
            'missing_values' : missing_values,
            'missing_percentage' : missing_percentage
})

# Show only columns that have missing value
missing_df = missing_df[missing_df['missing_values'] > 1]
if len(missing_df) > 1:
    print(f"Columns with missing values in RCB match Data: {missing_df.sort_values(by = 'missing_values', ascending = False)}")
else:
    print("No missing values found in the data")

Columns with missing values in RCB match Data:                  missing_values  missing_percentage
method                      252            0.988235
city                         14            0.054902
result_margin                 6            0.023529
winner                        3            0.011765
player_of_match               3            0.011765


In [97]:
# RCB's match analysis

In [98]:
# Calculate win percentage
rcb_wins = len(rcb_matches[rcb_matches['winner'] == My_Team])
rcb_losses = len(rcb_matches) - rcb_wins
win_percentage = (rcb_wins / len(rcb_matches)) * 100

print(f"RCB Wins: {rcb_wins}")
print(f"RCB Losses: {rcb_losses}")
print(f"Win Percentage: {win_percentage:.1f}%")

RCB Wins: 123
RCB Losses: 132
Win Percentage: 48.2%


## Day 1 Summary

- **Total RCB matches:** 255
- **Wins:** 123
- **Losses:** 132
- **Overall win percentage:** 48.2%
- **Seasons played:** 2008–2024
- **Missing data noted in:** method, city, result_margin, winner, player_of_match

Next: Season-by-season breakdown and win% trend chart.